modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/report/DDMFormFieldTypeReportProcessor.java

```diff
@@ -45,4 +46,20 @@ public default JSONObject process(
 		return process(ddmFormFieldValue, null, 0, ddmFormInstanceReportEvent);
 	}
 
+	public default void updateData(
```

Comment:
@rodrigopaulino, can you please give more details why did you decide to move this logic to the interface?

Will the method `updateData` be necessary on `NumericDDMFormFieldTypeReportProcessor` and `TextDDMFormFieldTypeReportProcessor`?

I'm asking this because the method `process` is a method which necessarily needs to be implemented by every field processor used in Form Reports, e.g. the data displayed inside Entries Tab, therefore Marcos has decided to create the interface and include the method `process` for this reason.

Did you intend to reuse logic with this commit? If so, which possibilities did you have in mind to reuse logic despite of adding a default implementation in the interface?

Reply:
Hey, @natocesarrego. The `updateData` is not necessary on the Numeric and Text report processors. This is just common code that was extracted from the Grid, Radio and Checkbox report processors.
It's extracted as a default method in the interface because that's the code convention. Please, check [this](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752634/API+conventions+in+Liferay#%5BinlineExtension%5DUtil-should-mirror-the-interface).

Hope this clarifies. Let me know in case you need further details. 😄

Reply:
Thanks for the clarification! I just saw one of the references mentioned by the doc shared by you. :wink: